### PR TITLE
Added Azure key vault CSI driver

### PR DESCRIPTION
**Pull request by satr**

### CODE REVIEW

File: clusters/development/kustomization.yaml [hunk 1]
```diff
@@ -37,6 +37,7 @@ patches:
   - path: ./overlay/radix-platform/radix-operator/helmRelease.yaml
   - path: ./overlay/radix-platform/radix-velero-plugin/postBuild.yaml
   - path: ./overlay/third-party/blob-csi-driver/helmRelease.yaml
+  - path: ./overlay/third-party/csi-secrets-store-provider-azure/helmRelease.yaml
   - path: ./overlay/third-party/cert-manager/helmRelease.yaml
```
Comment (oleterno-equinor) [inline clusters/development/kustomization.yaml:41]: alphabetical sorting makes it easier to read
```suggestion
  - path: ./overlay/third-party/cert-manager/helmRelease.yaml
  - path: ./overlay/third-party/csi-secrets-store-provider-azure/helmRelease.yaml
```

File: clusters/production/kustomization.yaml [hunk 2]
```diff
@@ -4,6 +4,7 @@ resources:
 - flux-system
 - ./imageUpdateAutomation.yaml
 - ../../components/third-party/blob-csi-driver
+- ../../components/third-party/csi-secrets-store-provider-azure
```
Comment (oleterno-equinor) [inline clusters/production/kustomization.yaml:7]: should this be in the development overlay instead?

Comment (satr) [reply clusters/production/kustomization.yaml]: Yes - this is a mistake. Fixed.

File: clusters/development/overlay/third-party/csi-secrets-store-provider-azure/helmRelease.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: csi-secrets-store-provider-azure
+  namespace: kube-system
+spec:
+  chart:
+    spec:
+      version: v1.0.0 # https://github.com/kubernetes-sigs/blob-csi-driver/releases
```
Comment (oleterno-equinor) [inline clusters/development/overlay/third-party/csi-secrets-store-provider-azure/helmRelease.yaml:9]:
```suggestion
      version: v1.0.0 # https://github.com/Azure/secrets-store-csi-driver-provider-azure/releases
```

File: clusters/playground/overlay/third-party/csi-secrets-store-provider-azure/helmRelease.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: csi-secrets-store-provider-azure
+  namespace: kube-system
+spec:
+  chart:
+    spec:
+      version: v1.0.0 # https://github.com/kubernetes-sigs/blob-csi-driver/releases
```
Comment (oleterno-equinor) [inline clusters/playground/overlay/third-party/csi-secrets-store-provider-azure/helmRelease.yaml:9]:
```suggestion
      version: v1.0.0 # https://github.com/Azure/secrets-store-csi-driver-provider-azure/releases
```

File: clusters/production/overlay/third-party/csi-secrets-store-provider-azure/helmRelease.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: csi-secrets-store-provider-azure
+  namespace: kube-system
+spec:
+  chart:
+    spec:
+      version: v1.0.0 # https://github.com/kubernetes-sigs/blob-csi-driver/releases
```
Comment (oleterno-equinor) [inline clusters/production/overlay/third-party/csi-secrets-store-provider-azure/helmRelease.yaml:9]:
```suggestion
      version: v1.0.0 # https://github.com/Azure/secrets-store-csi-driver-provider-azure/releases
```

**Review by oleterno-equinor — APPROVED** [review]

**Pull request merged**